ome_zarr/writer.py

```diff
@@ -302,6 +315,7 @@ def write_multiscales_metadata(
         dict(
             version=fmt.version,
             datasets=_validate_datasets(datasets, ndim, fmt),
+            name=name if name else group.name,
```

Comment:
this line might be too heavyhanded, as the spec says that name "SHOULD" be provided but not "MUST".  Is it ok for this code to write out group.name?  I believe there were tests that depended on name having been written.